lib/themes/dosomething/paraneue_dosomething/js/social-auth.js

```diff
@@ -0,0 +1,28 @@
+const $ = require('jquery');
+
+function checkFacebookLoginState() {
+  FB.getLoginStatus((response) => {
+    switch (response.status) {
+      case 'connected': console.log('Logged in'); break;
+      case 'not_authorized': console.log('App needs authorization'); break;
+      default: console.log("not logged into facebook");
+    }
+  });
+}
+// Attaching this to window so the Facebook element can call it without scope issues.
+window.checkFacebookLoginState = checkFacebookLoginState;
+
+$(document).ready(function() {
+  window.fbAsyncInit = function() {
+    FB.init({
+      appId      : '105775762330',
+      xfbml      : true,
+      version    : 'v2.7'
+    });
+
+    checkFacebookLoginState();
```

Comment:
To summarize an IRL conversation: this will automatically register/login a user on page load via their Facebook ID, as long as they have an open Facebook session in their browser. This seems like it would prevent users who do _not_ have Facebook accounts linked on DoSomething from ever logging in to "classic" accounts without first logging out of Facebook.